src/download.ts

```diff
@@ -1,79 +1,79 @@
 import { sync as mkdirpSync } from "mkdirp";
 import fs from "fs";
 import shell from "shelljs";
-
+import chalk from "chalk";
 import config from "./config";
-
-import { updateFile, exportFile } from "./utils/client";
+import {
+  updateOrRestoreFile,
+  exportFile,
+  isCommonErrorResponse,
+} from "./lib/crowdin";
 import log from "./utils/logging";
-import convertToKeyValue from "./utils/convertToKeyValue";
+import convertToKeyValue from "./utils/convert-to-key-value";
 
 export default async (isTS = false) => {
-  /* sync source file to prevent branching issues */
-
+  // Sync source file to prevent branching issues
   if (!fs.existsSync(config.TRANSLATIONS_FILE)) {
     log.error(`${config.TRANSLATIONS_FILE} does not exist.`);
     process.exit(1);
   }
 
-  log.info("CROWDIN", "Starting download process...");
+  log.info("Syncing source file");
 
   mkdirpSync(config.TRANSLATIONS_DIR);
 
   const file = fs.createReadStream(config.TRANSLATIONS_FILE);
   const fileExtension = isTS ? "ts" : "js";
+  const updateResponse = await updateOrRestoreFile(config.BRANCH_NAME, file);
 
-  const updateResponse = await updateFile(config.BRANCH, file);
-
-  log.info("", "Syncing source.json");
-
-  if (!updateResponse?.data?.success) {
-    const { error } = updateResponse.data;
+  if (isCommonErrorResponse(updateResponse)) {
     log.error(
-      error?.message || "Something went wrong when uploading source.json"
+      updateResponse.error.message ||
+        `Something went wrong while uploading the source file`
     );
     process.exit(1);
   }
 
-  log.info("", "Downloading translations...");
+  log.info("Downloading translations from Crowdin");
 
   const exportFileResponses = await Promise.all(
-    config.CROWDIN_LANGUAGES.map(language =>
-      exportFile(config.BRANCH, language)
-    )
+    config.CROWDIN_LANGUAGES.map((language) => {
+      return exportFile(config.BRANCH_NAME, language);
+    })
   );
 
-  log.info("", `Writing translations to: ${config.TRANSLATIONS_DIR}`);
+  log.info(`Writing translations to: ${chalk.bold(config.TRANSLATIONS_DIR)}`);
 
   await Promise.all(
     config.CROWDIN_LANGUAGES.map((language, i) => {
-      const translations = exportFileResponses[i].data;
-      if (translations.error) {
+      const response = exportFileResponses[i];
+
+      if (!response || response.data.error) {
```

Comment:
This was causing issues in v1 already. If there is a string with an `error` key the scripts will stop here. Maybe it isn't for the context of this PR, so we should work on a follow-up version that handles HTTP errors in a different way (throw an error instead of resolving with an `error` key that could conflict with a translation string).

Reply:
maybe we can also check the returned statuscode?

Reply:
Yep, indeed. Checking for the status code now. Tried to reproduce all edge-cases in the mobile app project and seems resolved.